### PR TITLE
[030] added Query methods

### PR DESCRIPTION
**Pull request by Maksim-maevsky**
- added scripts to user_skill, internship_skill
 -  added Query methods for fetch type lazy

### CODE REVIEW

File: src/main/java/by/exadel/internship/service/UserService.java [hunk 1]
```diff
@@ -18,7 +20,7 @@
 
     public List<UserDTO> getAll() {
 
-        List<User> userList = userRepository.findAll();
+        List<User> userList = userRepository.findAllWithSkill();
```
Comment (Fenydude) [inline src/main/java/by/exadel/internship/service/UserService.java]: Add logging to this method

Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/UserService.java]: done

**Review by ShirshovAlex — DISMISSED** [review]

File: src/main/java/by/exadel/internship/service/FormService.java [hunk 2]
```diff
@@ -67,14 +81,38 @@ private void uploadFile(MultipartFile file, UUID uuid) {
...
-        return formRepository.findAll();
+    public List<FormFullDTO> getAll() {
+
+        MDC.put("className", FormService.class.getSimpleName());
+        log.info("Try to get all forms");
+
+        List<Form> formList = formRepository.findAllWithTimeForCallList();
+
+        System.out.println();
+        for (Form f:formList) {
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/service/FormService.java]: Remove it at all or do via logger

Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/FormService.java]: yaa, I forgot about it, thanks, it was for test. Done.

File: src/main/java/by/exadel/internship/service/InternshipService.java [hunk 3]
```diff
@@ -22,29 +22,43 @@
...
 
         MDC.put("className", InternshipService.class.getSimpleName());
-        log.info("Try to get all Internships");
+        log.info("Try to get all Internships with skills");
+
+        List<Internship> internships = internshipRepository.findAllWithSkill();
+
+        System.out.println();
+        for (Internship i:internships) {
+            System.out.println(i.getId());
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/service/InternshipService.java]: Why do you use System.out.println?

Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/InternshipService.java]: i removed it

File: src/main/java/by/exadel/internship/service/FormService.java [hunk 4]
```diff
@@ -36,45 +36,75 @@
...
         return mapper.toFormDto(form);
+
     }
 
-    private void uploadFile(MultipartFile file, UUID uuid) {
+    private void uploadFile(MultipartFile file, UUID id) {
+
         try {
-            Path path = Paths.get(filePath + File.separator + uuid);
+            Path path = Paths.get(filePath + File.separator + id);
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/FormService.java]: Use Paths.get(filePath, id) instead

File: src/main/java/by/exadel/internship/service/FormService.java [hunk 5]
```diff
@@ -36,45 +36,75 @@
...
-    public List<Form> getAll() {
-        return formRepository.findAll();
+    public List<FormFullDTO> getAll() {
+
+        MDC.put("className", FormService.class.getSimpleName());
+        log.info("Try to get all forms");
+
+        List<Form> formList = formRepository.findAllWithTimeForCallList();
+
+        log.info("Successfully list of forms");
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/FormService.java]: Double log message looks redundant

File: src/main/java/by/exadel/internship/service/InternshipService.java [hunk 6]
```diff
@@ -22,29 +22,35 @@
     private final InternshipRepository internshipRepository;
     private final InternshipMapper internShipMapper;
 
-    public GuestInternshipDTO getById(UUID uuid) {
+    public GuestInternshipDTO getById(UUID id) {
 
         MDC.put("className", InternshipService.class.getSimpleName());
-        log.info("Try to get Internships with uuid=" + uuid);
+        log.info("Try to get Internships with id=" + id);
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/InternshipService.java]: Why not using substitution approach?

log.info("Try to get Internships with id={}", id);

File: src/main/java/by/exadel/internship/service/UserService.java [hunk 7]
```diff
@@ -1,34 +1,57 @@
...
 
     public List<UserDTO> getAll() {
 
-        List<User> userList = userRepository.findAll();
+        MDC.put("className", UserService.class.getSimpleName());
+        log.info("Try to get all users with skill");
+
+        List<User> userList = userRepository.findAllWithSkill();
+
+        log.info("Successfully list of users");
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/service/UserService.java]: Double log message looks redundant

**Review by SergeyKoval — CHANGES_REQUESTED** [review]

File: src/main/java/by/exadel/internship/service/FormService.java (hunk 4 above)
Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/FormService.java]: done

File: src/main/java/by/exadel/internship/service/FormService.java (hunk 5 above)
Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/FormService.java]: done

File: src/main/java/by/exadel/internship/service/InternshipService.java (hunk 6 above)
Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/InternshipService.java]: done

File: src/main/java/by/exadel/internship/service/UserService.java (hunk 7 above)
Comment (Maksim-maevsky) [reply src/main/java/by/exadel/internship/service/UserService.java]: done

**Review by SergeyKoval — APPROVED** [review]

**Review by ShirshovAlex — APPROVED** [review]

**Pull request merged**